homeassistant/components/rainforest_raven/config_flow.py

```diff
@@ -0,0 +1,156 @@
+from serial.tools.list_ports_common import ListPortInfo
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import usb
+from homeassistant.const import CONF_DEVICE, CONF_MAC, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_NAME, DOMAIN
+
+
+def _format_id(value: str | int) -> str:
+    if isinstance(value, str):
+        return value
+    return f"{value or 0:04X}"
+
+
+def _generate_unique_id(info: ListPortInfo | usb.UsbServiceInfo) -> str:
+    """Generate unique id from usb attributes."""
+    return f"{_format_id(info.vid)}:{_format_id(info.pid)}_{info.serial_number}_{info.manufacturer}_{info.description}"
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rainforest RAVEn devices."""
+
+    def __init__(self) -> None:
+        """Set up flow instance."""
+        self._dev_path: str | None = None
+        self._meter_macs: dict[bytes, str] = {}
+
+    async def _validate_device(self, dev_path: str) -> None:
+        self._abort_if_unique_id_configured(updates={CONF_DEVICE: dev_path})
+        async with (
+            async_timeout.timeout(5),
+            RAVEnSerialDevice(dev_path) as raven_device,
+        ):
+            await raven_device.synchronize()
+            meters = await raven_device.get_meter_list()
+            for meter in meters.meter_mac_ids:
+                meter_info = await raven_device.get_meter_info(meter=meter)
+                self._meter_macs[meter] = (
+                    "unknown"
+                    if meter_info.meter_type is None
+                    else str(meter_info.meter_type)
+                )
+        self._dev_path = dev_path
+
+    async def async_step_meters(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Connect to device and discover meters."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            meter_macs = []
+            for raw_mac in user_input.get(CONF_MAC, "").split(","):
+                try:
+                    mac = bytes.fromhex(raw_mac)
+                except ValueError:
+                    errors[CONF_MAC] = "invalid_mac"
+                    break
+                if len(mac) != 8:
+                    errors[CONF_MAC] = "invalid_mac"
+                    break
+                if self._meter_macs.get(mac) not in ("unknown", "electric"):
+                    errors[CONF_MAC] = "no_meters_found"
+                    break
+                if mac not in meter_macs:
+                    meter_macs.append(mac)
+            if meter_macs and not errors:
+                return self.async_create_entry(
+                    title=user_input.get(CONF_NAME, DEFAULT_NAME),
+                    data={
+                        CONF_DEVICE: self._dev_path,
+                        CONF_MAC: [m.hex() for m in meter_macs],
+                    },
+                )
```

Comment:
Can be simplified if using a SelectSelector

Reply:
I believe that `SelectSelector` means that only a single value can be chosen, right? The RAVEn can be connected to multiple meters (though I've never tested it).

If you're okay with that limitation, I can switch to the `SelectSelector` anyway.

Reply:
You need to set the `multiple` flag to `True` on the `SelectSelector`

Reply:
Thanks, @emontnemery.

This was implemented in bdce69a.